### PR TITLE
IOOS  1.2  platform  DSG

### PR DESCRIPTION
**Pull request by daltonkell**
Separate platform checks from the CF DSG checks and add corresponding unit tests, per #748.

### CODE REVIEW

File: compliance_checker/ioos.py [hunk 1]
```diff
@@ -815,62 +806,114 @@ def check_single_platform(self, ds):
...
+                    "profile restricts {feature_type} datasets to a "
+                    "single platform (ie. station/trajectory/profile) per dataset."
+                ).format(
+                    cf_role_var=var.name, cf_role=cf_role,
+                    dim=shp, feature_type=feature_type
+                )
+ 
+                _val = shp==1
+ 
+            elif feature_type=="point": # do nothing
```
Comment (benjwadams) [inline compliance_checker/ioos.py:900]: Please leave spaces around binary operators and non-keyword assignments per suggestions in PEP8.

Comment (Bobfrat) [reply compliance_checker/ioos.py:900]: if you're going to enforce pep8, might be worth letting travisCI catch those violations. Likewise, @daltonkell there are tools you can use that provide a pre-commit hook to check your code for pep8 compliance.

Comment (benjwadams) [reply compliance_checker/ioos.py:900]: @Bobfrat, The thing is with CI is that a lot of the existent codebase is not necessarily PEP8 compliant already.  We have a "coding-standards" test in Travis already which at a cursory glance appears to check PEP8 compliance:  https://travis-ci.org/github/ioos/compliance-checker/jobs/672560182

It allows for failure, but to my knowledge it's running across the entire code base, so we'd have to do some real refactoring of the code base if we didn't want that test to fail and to provide valuable input.

Thus, until we push our code through something like `black` and refactor, I'd prefer to handle this on the coding side.

Comment (Bobfrat) [reply compliance_checker/ioos.py:900]: Not sure what you had in mind for refactoring but you can get the code base a pretty good ways to pep8 compliant with autopep8. I am definitely not an autopep8 expert but the couple times I've used it I was happy with the results. 

```
find ./compliance_checker -name \*.py -exec autopep8 --recursive --aggressive --in-place {} +
```

Comment (benjwadams) [reply compliance_checker/ioos.py]: @ocefpaf might know more about some of the tooling here.  Perhaps we should open a new issue to address coding standards/PEP8, etc.

Comment (ocefpaf) [reply compliance_checker/ioos.py]: Do you want to use `pre-commit` hooks do to this? I can setup GH actions as a CI and those who have the pre-commit locally will get the code adjustments "for free." I'll open a PR to demonstrate that.

**Pull request merged**